tests/test_estimators.py

```diff
@@ -48,55 +48,48 @@ def get_kneighbor_estimator_instances() -> List[IDNeighborsRegressor]:
 
 
 @pytest.mark.parametrize("estimator", get_kneighbor_estimator_classes())
-def test_estimators_raise_notfitted_kneighbors(estimator, moscow_euclidean):
+def test_estimators_raise_notfitted_kneighbors(estimator):
     """Attempting to call kneighbors on an unfitted estimator should raise."""
+    X, y = load_moscow_stjoes(return_X_y=True)
     with pytest.raises(NotFittedError):
-        estimator().kneighbors(moscow_euclidean.X)
+        estimator().kneighbors(X)
 
 
 @pytest.mark.parametrize("estimator", get_kneighbor_estimator_classes())
-def test_estimators_raise_notfitted_predict(estimator, moscow_euclidean):
+def test_estimators_raise_notfitted_predict(estimator):
     """Attempting to call predict on an unfitted estimator should raise."""
+    X, y = load_moscow_stjoes(return_X_y=True)
     with pytest.raises(NotFittedError):
-        estimator().predict(moscow_euclidean.X)
+        estimator().predict(X)
 
 
 @pytest.mark.parametrize("estimator", get_kneighbor_estimator_classes())
-def test_estimators_support_continuous_multioutput(estimator, moscow_euclidean):
+def test_estimators_support_continuous_multioutput(estimator):
     """All estimators should fit and predict continuous multioutput data."""
+    X, y = load_moscow_stjoes(return_X_y=True)
     estimator = estimator()
-    estimator.fit(moscow_euclidean.X, moscow_euclidean.y)
-    estimator.predict(moscow_euclidean.X)
+    estimator.fit(X, y)
+    estimator.predict(X)
 
 
 @pytest.mark.parametrize("with_names", [True, False])
 @pytest.mark.parametrize("estimator", get_kneighbor_estimator_classes())
-def test_estimators_support_dataframes(estimator, with_names, moscow_euclidean):
+def test_estimators_support_dataframes(estimator, with_names):
     """All estimators should fit and predict data stored as dataframes."""
-    estimator = estimator()
-    num_features = moscow_euclidean.X.shape[1]
-    feature_names = [f"col_{i}" for i in range(num_features)] if with_names else None
-
-    X_df = pd.DataFrame(moscow_euclidean.X, columns=feature_names)
-    y_df = pd.DataFrame(moscow_euclidean.y)
-
-    estimator.fit(X_df, y_df)
-    estimator.predict(X_df)
+    X, y = load_moscow_stjoes(return_X_y=True, as_frame=True)
+    estimator = estimator().fit(X, y)
+    estimator.predict(X)
 
-    assert_array_equal(getattr(estimator, "feature_names_in_", None), feature_names)
+    assert_array_equal(getattr(estimator, "feature_names_in_", None), X.columns)
```

Comment:
Being able to rewrite this test like this is a great benefit of the datasets refactor.